src/app/pages/statistics/shared/statistics-filters.component.ts

```diff
@@ -124,11 +143,50 @@ export class StatisticsFiltersComponent implements OnInit {
       }
     }
     this.setDateFilterYear();
+    this.setDateRangeFilterYear(true);
     this.filterParams = this.buildFilterValues();
     this.filters.emit(this.filterParams);
     this.update.emit(true);
   }
 
+  public dateRangeChange(filterDef: StatisticsFilterDef, event): void {
+    if (filterDef.type === 'date-range' && event.hasOwnProperty('startDate') && event.hasOwnProperty('endDate')) {
+      filterDef.currentValue = event ? event : null;
+    } else {
+      const dateRange = event.currentTarget.value;
```

Comment:
Why the else here?
Manual update by user?

Reply:
Yes for the manual update